.github/workflows/e2e_test.yml

```
@@ -0,0 +1,33 @@
+name: E2E Tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.23"
+
+      - uses: hetznercloud/tps-action@main
+
+      - name: Run tests
+        run: go test -tags e2e -coverpkg=./... -coverprofile=coverage.txt -v -race ./e2e_test
+
+      - name: Upload coverage reports to Codecov
+        if: >
+          !startsWith(github.head_ref, 'renovate/') &&
+          !startsWith(github.head_ref, 'release-please--')
+        uses: codecov/codecov-action@v4
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Needs flags so codecov can properly show that these are two different kinds of coverage. Please also add a `flags: unit` to the unit test coverage upload.

```suggestion
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          flags: e2e
```